docs/src/guide/entities_components_world.md

```diff
@@ -0,0 +1,111 @@
++++
+type = "docs"
+title = "Entities, components and the world"
+weight = 10
++++
+
+Larecs🌲 provides a flexible and efficient way to store, 
+access and manipulate data records organized 
+around individual entities. Suppose we wanted to 
+model a world with different objects and agents that each have 
+certain properties. The idea of entity component systems 
+(ECS) is to characterize these objects and agents 
+(called "entities" in an ECS) via the set of 
+attributes ("components") they possess, and to 
+model the interactions between them via "systems",
+functions that operate on entities 
+with certain components. Larecs🌲 provides the functionality 
+and tools to implement such entity component systems.
+
+## Entities
+
+Entities are the central unit around which the data in
+an ECS is organized. Entities could represent game objects,
+or potentially anything else that "lives" in a game or simulation.
+Each entity can possess an arbitrary
+set of components, which can also be used to characterize
+an entity. That is, in an ECS, cars would not directly 
+identified as "cars" but rather as everything that has 
+the components `Position`, `Velocity`, `Fuel reserves`, 
+`Engine power`, etc. 
+
+The components are not stored in individual 
+objects but rather in a central container, called the [`World`](#the-world). 
+Hence, an {{< api Entity >}} is merely an identifier that allows
+retrieving the corresponding components from the world.
+As such, entities are strictly bound to the world they live in,
+but also small in memory and easy to store.
+
+> [!NOTE] 
+> Though entities can safely be stored and passed around,
+> their components (or pointers to them) should never be stored
+> externally, as they can move in memory at any time.
+
+How entities are created and used is discussed in the
+[next chapters](../adding_and_removing_entities).
+
+## Components
+
+Components are the data associated with entities, characterizing
+their state and properties. Components are represented
+via structs: each component type is a different struct.
+For example, if we want to model a world in which 
+entities may have a position and a velocity, we need to 
+define a `Position` and a `Velocity` struct.
+
+```mojo {doctest="guide_entities_components_world" global=true}
+@value
+struct Position:
+    var x: Float64
+    var y: Float64
+
+@value
+struct Velocity:
+    var dx: Float64
+    var dy: Float64
+```
+
+These structs are not only used to store data. Their
+types are also used as unique identifiers for the components
+of an entity. That is, they have a similar role to 
+a key in a dictionary.
+
+How components can be associated with entities is described 
```

Comment:
IMO always writing "associated" makes things harder to understand. Although components are not stored in entities, I would prefer "can be added to entities".